File: tests/languages/java/ServiceTest.java
package io.appwrite.services;

import com.google.gson.Gson;
import org.junit.jupiter.api.Test;
import io.appwrite.Client;
import okhttp3.Response;
import java.util.List;
import java.io.IOException;
import java.util.Map;

public class ServiceTest {

    @Test
    public void test() throws IOException {
        Client client = new Client();
        Foo foo = new Foo(client);
        Bar bar = new Bar(client);
        General general = new General(client);

        client.addHeader("Origin", "http://localhost");
        client.setSelfSigned(true);

// Foo Tests

        Response response;
        response = foo.get("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = foo.post("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = foo.put("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = foo.patch("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = foo.delete("string", 123, List.of("string in array")).execute();
        printResponse(response);

// Bar Tests

        response = bar.get("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = bar.post("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = bar.put("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = bar.patch("string", 123, List.of("string in array")).execute();
        printResponse(response);

        response = bar.delete("string", 123, List.of("string in array")).execute();
        printResponse(response);

        // General Tests

        response = general.redirect().execute();
        printResponse(response);

        // response = await general.setCookie();
        // System.out.println(response.data["result"]);

        // response = await general.getCookie();
        // System.out.println(response.data["result"]);
    }

    private void printResponse(Response response) throws IOException {
        Gson gson = new Gson();
        Map map = gson.fromJson(response.body().string(), Map.class);
        System.out.println(map.get("result"));

    }
}

File: tests/languages/kotlin/ServiceTest.kt
package io.appwrite

import com.google.gson.Gson
import io.appwrite.exceptions.AppwriteException
import io.appwrite.extensions.fromJson
import io.appwrite.extensions.toJson
import io.appwrite.models.Error
import io.appwrite.models.InputFile
import io.appwrite.models.Mock
import io.appwrite.services.Bar
import io.appwrite.services.Foo
import io.appwrite.services.General
import kotlinx.coroutines.runBlocking
import okhttp3.Response
import org.junit.Before
import org.junit.Test
import java.io.File
import java.io.IOException
import java.nio.file.Files
import java.nio.file.Paths

class ServiceTest {

    private val filename: String = "result.txt"

    @Before
    fun setUp() {
        Files.deleteIfExists(Paths.get(filename))
        writeToFile("Test Started")
    }

    @Test
    @Throws(IOException::class)
    fun test() {
        val client = Client()
            .addHeader("Origin", "http://localhost")
            .setSelfSigned(true)
        val foo = Foo(client, "string")
        val bar = Bar(client)
        val general = General(client)

        runBlocking {
            var mock: Mock
            // Foo Tests
            mock = foo.get(123, listOf("string in array"))
            writeToFile(mock.result)
            mock = foo.post(123, listOf("string in array"))
            writeToFile(mock.result)
            mock = foo.put(123, listOf("string in array"))
            writeToFile(mock.result)
            mock = foo.patch(123, listOf("string in array"))
            writeToFile(mock.result)
            mock = foo.delete(123, listOf("string in array"))
            writeToFile(mock.result)

            // Bar Tests
            mock = bar.get("string", 123, listOf("string in array"))
            writeToFile(mock.result)
            mock = bar.post("string", 123, listOf("string in array"))
            writeToFile(mock.result)
            mock = bar.put("string", 123, listOf("string in array"))
            writeToFile(mock.result)
            mock = bar.patch("string", 123, listOf("string in array"))
            writeToFile(mock.result)
            mock = bar.delete("string", 123, listOf("string in array"))
            writeToFile(mock.result)

            // General Tests
            val result = general.redirect()
            writeToFile((result as Map<String, Any>)["result"] as String)

            try {
                mock = general.upload("string", 123, listOf("string in array"), InputFile.fromPath("../../resources/file.png"))
                writeToFile(mock.result)
            } catch (ex: Exception) {
                writeToFile(ex.toString())
            }
            try {
                mock = general.upload("string", 123, listOf("string in array"), InputFile.fromPath("../../resources/large_file.mp4"))
                writeToFile(mock.result)
            } catch (ex: Exception) {
                writeToFile(ex.toString())
            }
            try {
                var bytes = File("../../resources/file.png").readBytes()
                mock = general.upload("string", 123, listOf("string in array"), InputFile.fromBytes(bytes, "file.png", "image/png"))
                writeToFile(mock.result)
            } catch (ex: Exception) {
                writeToFile(ex.toString())
            }
            try {
                var bytes = File("../../resources/large_file.mp4").readBytes()
                mock = general.upload("string", 123, listOf("string in array"), InputFile.fromBytes(bytes, "large_file.mp4", "video/mp4"))
                writeToFile(mock.result)
            } catch (ex: Exception) {
                writeToFile(ex.toString())
            }

            val res = general.download()
            writeToFile(String(res, Charsets.UTF_8))

            try {
                general.error400()
            } catch (e: AppwriteException) {
                writeToFile(e.message)
            }

            try {
                general.error500()
            } catch (e: AppwriteException) {
                writeToFile(e.message)
            }

            try {
                general.error502()
            } catch (e: AppwriteException) {
                writeToFile(e.message)
            }

            general.empty()
        }
    }

    private fun writeToFile(string: String?) {
        val text = "${string ?: ""}\n"
        File("result.txt").appendText(text)
    }

}